client-app/core/composables/useGoogleAnalytics.ts

```diff
@@ -17,9 +21,46 @@ type CustomEventNamesType = "place_order" | "clear_cart";
 type EventParamsType = Gtag.ControlParams & Gtag.EventParams & Gtag.CustomParams;
 type EventParamsExtendedType = EventParamsType & { item_list_id?: string; item_list_name?: string };
 
-const { storeSettings } = useAppContext();
+const DEBUG_PREFIX = "[GA]";
 
-const isAvailableGtag: Readonly<boolean> = Boolean(IS_CLIENT && storeSettings.googleAnalyticsEnabled && window.gtag);
+const MODULE_KEYS = {
+  ID: "VirtoCommerce.GoogleEcommerceAnalytics",
+  ENABLE_STATE: "GoogleAnalytics4.EnableTracking",
+  TRACK_ID: "GoogleAnalytics4.MeasurementId",
+};
+
+const { isThemeContextReady, modulesSettings } = useThemeContext();
+
+watch(isThemeContextReady, (isReady: boolean) => {
```

Comment:
I think it's better to explicitly name it `initialize` and call from `app-runner` because now it's calling only because it used in components which accidentally put into main bundle. If we will optimize bunle, this will not working if we will open page where `useGoogleAnalytics` is not used until we open page where it used.

Reply:
ok